mysql/resource_grant.go

```diff
@@ -19,16 +21,25 @@ func resourceGrant() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"user": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"role"},
+			},
+
+			"role": &schema.Schema{
```

Comment:
Can someone have both `role` and `roles` attribute defined in the HCL?

Reply:
I don't believe you can assign `roles` to a `role`.